### PR TITLE
fix: resolve sonar issues

### PR DESCRIPTION
**Pull request by ivan-kalachikov**
### Jira-link:
https://virtocommerce.atlassian.net/browse/VCST-1671
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-2.3.0-pr-1252-c46d-c46d3b69.zip

### CODE REVIEW

File: client-app/core/api/graphql/push-messages/mutations/clearAllPushMessages/index.ts [hunk 1]
```diff
@@ -9,7 +9,6 @@ export function useClearAllPushMessages() {
     optimisticResponse: {
       clearAllPushMessages: true,
     },
-    // TODO: Refactor updateQueries to use update since it will be deprecated in the next version of Apollo Client - https://www.apollographql.com/docs/react/api/react/hoc/#optionsupdatequeries
```
Comment (ivan-kalachikov) [inline client-app/core/api/graphql/push-messages/mutations/clearAllPushMessages/index.ts:12]: The according task has been created: https://virtocommerce.atlassian.net/browse/VCST-1674

File: client-app/core/utilities/address/index.ts [hunk 2]
```diff
@@ -2,7 +2,6 @@ import { clone, isEqual, pick } from "lodash";
 import type { AnyAddressType } from "../../types";
 import type { InputMemberAddressType, MemberAddressType } from "@/core/api/graphql/types";
 
-// FIXME: https://virtocommerce.atlassian.net/browse/ST-5122
```
Comment (ivan-kalachikov) [inline client-app/core/utilities/address/index.ts:5]: the task is cancelled, so FIXME is not relevant

File: client-app/pages/account/edit-quote.vue [hunk 3]
```diff
@@ -422,7 +422,7 @@ async function submit(): Promise<void> {
 
   await submitQuote(quote.value!.id, comment.value || "");
 
-  void router.replace({ name: "Quotes" });
+  router.replace({ name: "Quotes" }).catch(Logger.error);
```
Comment (NaMax66) [inline client-app/pages/account/edit-quote.vue:425]: return void. If no need result we place it before promise.

File: client-app/shared/layout/components/search-bar/search-bar.vue [hunk 4]
```diff
@@ -286,15 +286,15 @@ function reset() {
...
-    void searchAndShowDropdownResults();
+    await searchAndShowDropdownResults();
   }
 }, SEARCH_BAR_DEBOUNCE_TIME);
 
-function onSearchPhraseChanged() {
+async function onSearchPhraseChanged() {
   hideSearchDropdown();
-  void searchProductsDebounced();
+  await searchProductsDebounced();
```
Comment (NaMax66) [inline client-app/shared/layout/components/search-bar/search-bar.vue:297]: return void everywhere and add rule for sonar to accept it.
It needs to fix the warning
image

**Review by NaMax66 — CHANGES_REQUESTED** [review]

File: client-app/pages/account/edit-quote.vue (hunk 3 above)
Comment (ivan-kalachikov) [reply client-app/pages/account/edit-quote.vue]: reverted changes with "void"

File: client-app/shared/layout/components/search-bar/search-bar.vue (hunk 4 above)
Comment (ivan-kalachikov) [reply client-app/shared/layout/components/search-bar/search-bar.vue]: reverted changes with "void"

**Review by NaMax66 — APPROVED** [review]

**Pull request merged**